### ISSUE TITLE
WASM Support

### ISSUE THREAD

**Issue by AndrewJSchoen** [alphaville/optimization-engine#248]
**Is your feature request related to a problem? Please describe.**
I am working on a project that uses this crate, and we are looking to deploy the project in WebAssembly for use on the web. However, it appears that this library is not currently setup to handle this, and as such it prevents this from being possible as-is. 

**Describe the solution you'd like**
I know of [one fork of this repo](https://github.com/yepw/optimization-engine) that has added this support. Since it would make sense for such a change to be as minimally invasive as possible, it might make sense to configure such implementation as a feature that could be enabled.

**Describe alternatives you've considered**
Currently I am just using the other repo mentioned, but it would be nice to see this in the official repo/crate.

**Comment by alphaville** [comment]
Thank you, @AndrewJSchoen for reporting this. Would it be convenient if we supported WebAssembly using a [feature](https://doc.rust-lang.org/cargo/reference/features.html)?

**Comment by AndrewJSchoen** [comment]
Yes, that would be fantastic!

**Comment by alphaville** [comment]
@AndrewJSchoen  Support for Wasm has been added to OpEn version [0.7.4](https://crates.io/crates/optimization_engine/0.7.4) - just released. You can have a look at [the doucmentation](https://alphaville.github.io/optimization-engine/docs/openrust-features#webassembly-support).

**Comment by AndrewJSchoen** [comment]
This looks great! Thanks!

**Comment by AndrewJSchoen** [comment]
I was working with the new WASM support, and specified it my cargo.toml as such:
```
optimization_engine = { version = "0.7.4", features = ["wasm"] }
```

, and encountered an error when running it. The only log I was able to retrieve from WebAssembly was 
```
panicked at 'time not implemented on this platform', library/std/src/sys/wasm/../unsupported/time.rs:13:9
```

I tested this again with the old version I was using ([yepw/optimization-engine](https://github.com/yepw/optimization-engine)), and this error did not appear. Maybe something didn't quite make it through to 0.7.4?

**Comment by alphaville** [comment]
@AndrewJSchoen thank you for reporting this - let me check.

**Comment by AndrewJSchoen** [comment]
Sounds good. Feel free to let me know if you would like me to test/test anything on my end!

**Comment by AndrewJSchoen** [comment]
@alphaville Any updates on this? Ideally I would like to switch to the official release.

**Comment by alphaville** [comment]
@AndrewJSchoen  This has been resolved in [OpEn v0.7.7](https://crates.io/crates/optimization_engine) which has been released.
